B2r2

briling opened the pull request:
- optimize b2r2
- add tests for b2r2 and slatm_d
- refactor 3D data loaders
- make xtb subsets of gdb and cyclo identical to equireact
- add CLI option `--xtb_subset` for `src/run_all_fingerprints.py`
- fix a gaussian kernel bug

briling committed:
Try to speedup B2R2_l

Replace scipy.stats.skewnorm.pdf() by np.exp()*(1+scipy.special.erf())
and rewrite the loop to be more clear

briling committed:
Fixup dece64c977: actually replace skewnorm

briling committed:
Remove misleading extra variable

briling committed:
Cleanup

briling committed:
Do the same for B2R2_n

briling committed:
Speed up B2R2_l by evaluating Gaussian one time for both bags

briling committed:
Repeat 1a8916eba1 for B2R2_n

briling committed:
Refactor B2R2_a (×2 speedup)

briling committed:
Refactor b2r2 wrappers

briling committed:
Refactor B2R2 wrappers. Found a bug

briling committed:
Critical bug fix in case representation is computed for less elements than in the dataset

briling committed:
Fixup 832fd9703c

briling committed:
Try to make a universal data loader (GDB done)

briling committed:
Continue d5e6865875 for cyclo

briling committed:
Universal data loader for proparg

briling committed:
Change xyz from custom to ase

briling committed:
Cleanup

puckvg reviewed (changes requested):
- The bad indices should be the same as for equireact, rather than having a different set of bad indices (confusing) 
- For B2R2 it's really hard for me to see whether the changes are ok, I think the best would be to have a couple of unit tests which just check that the initial and final functions yield the same results (eg for some small molecules that are in ase)

briling committed:
Fix bugs

- take the correct barriers when exclude bad indices
- actually use the hyperparameters for b2r2 @ gdb xtb
- add tqdm so it's less boring
- representation files norm diffs are ~1e-10 for the new b2r2_l
  implementation

briling committed:
Add tests for B2R2s and SLATM_d

briling committed:
Fix bug preventing exclusion of bad idx for gdb xtb subset @ dft geoms

briling committed:
Add (xtb_bad_idx + excluded rxns) from EquiReact

Now the xtb subsets for gdb and cyclo should be
identical to EquiReact

briling committed:
Add CLI option --xtb_subset for src/run_all_fingerprints.py

briling commented:
> * The bad indices should be the same as for equireact, rather than having a different set of bad indices (confusing)

Now the xtb subsets are the same for benchmark and equireact (the "normal" sets are still sligtly different).
 
> * For B2R2 it's really hard for me to see whether the changes are ok, I think the best would be to have a couple of unit tests 
which just check that the initial and final functions yield the same results (eg for some small molecules that are in ase)

Added units tests.

Also fixed bugs introduced above and for `src/run_all_fingerprints.py`  added a CLI option `--xtb_subset`
that uses DFT geometries at the xtb subset

briling committed:
Fix a bug introduced in 9bb00c5

For Gaussian kernel the grid was not consistent with
how the exponential was computed

briling commented:
> * The bad indices should be the same as for equireact, rather than having a different set of bad indices (confusing)

Now the xtb subsets are the same for benchmark and equireact (the "normal" sets are still sligtly different).
 
> * For B2R2 it's really hard for me to see whether the changes are ok, I think the best would be to have a couple of unit tests 
which just check that the initial and final functions yield the same results (eg for some small molecules that are in ase)

Added units tests.

Also:
- fixed a gaussian kernel bug introduced in 9bb00c5
- fixed bugs introduced above
- for `src/run_all_fingerprints.py`  added a CLI option `--xtb_subset`
that uses DFT geometries at the xtb subset
